// @aws-sdk/client-iam/dist/es/commands/CreateServiceSpecificCredentialCommand.js
import { __extends } from "tslib";
import { CreateServiceSpecificCredentialRequest, CreateServiceSpecificCredentialResponse } from "../models/models_0";
import { deserializeAws_queryCreateServiceSpecificCredentialCommand, serializeAws_queryCreateServiceSpecificCredentialCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Generates a set of credentials consisting of a user name and password that can be used
 *             to access the service specified in the request. These credentials are generated by
 *             IAM, and can be used only for the specified service. </p>
 *         <p>You can have a maximum of two sets of service-specific credentials for each supported
 *             service per user.</p>
 *         <p>You can create service-specific credentials for AWS CodeCommit and Amazon Keyspaces (for Apache
 *             Cassandra).</p>
 *         <p>You can reset the password to a new service-generated value by calling <a>ResetServiceSpecificCredential</a>.</p>
 *         <p>For more information about service-specific credentials, see <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_ssh-keys.html">Using IAM
 *                 with AWS CodeCommit: Git credentials, SSH keys, and AWS access keys</a> in the
 *                 <i>IAM User Guide</i>.</p>
 * @example
 * Use a bare-bones client and the command you need to make an API call.
 * ```javascript
 * import { IAMClient, CreateServiceSpecificCredentialCommand } from "@aws-sdk/client-iam"; // ES Modules import
 * // const { IAMClient, CreateServiceSpecificCredentialCommand } = require("@aws-sdk/client-iam"); // CommonJS import
 * const client = new IAMClient(config);
 * const command = new CreateServiceSpecificCredentialCommand(input);
 * const response = await client.send(command);
 * ```
 *
 * @see {@link CreateServiceSpecificCredentialCommandInput} for command's `input` shape.
 * @see {@link CreateServiceSpecificCredentialCommandOutput} for command's `response` shape.
 * @see {@link IAMClientResolvedConfig | config} for command's `input` shape.
 *
 */
var CreateServiceSpecificCredentialCommand = /** @class */ (function (_super) {
    __extends(CreateServiceSpecificCredentialCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function CreateServiceSpecificCredentialCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    CreateServiceSpecificCredentialCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IAMClient";
        var commandName = "CreateServiceSpecificCredentialCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateServiceSpecificCredentialRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateServiceSpecificCredentialResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateServiceSpecificCredentialCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateServiceSpecificCredentialCommand(input, context);
    };
    CreateServiceSpecificCredentialCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateServiceSpecificCredentialCommand(output, context);
    };
    return CreateServiceSpecificCredentialCommand;
}($Command));
export { CreateServiceSpecificCredentialCommand };
